homeassistant/components/reolink/services.yaml

```diff
@@ -16,3 +16,28 @@ ptz_move:
           min: 1
           max: 64
           step: 1
+
+chime_play:
+  target:
+    entity:
+      integration: reolink
+      domain: select
```

Comment:
Why is the target domain a select entity? Wouldn't it be more logical to have a device target?

Reply:
Because I absolutely hate using a device as a target because that will use some random device_id in the call.
When I then want to swap the device later on (because it is broke and I have a replacement, I bricked the firmware, I upgraded to a simular model with more features etc.) I just remove the old device and set the entity_ids of the new device to the same as the old device.

I get back all the history of the old device in the new device enitities, but most importantly all my automations will continue to work as usual since the entity_id is used as target.

Otherwise I would have to go trough my 195 automations and 102 scripts to check all the places where I used that device and re-target them to the new device. (ending up with searching trough YAML files for the old device_id, which is not user friendly).

Reply:
Right, but that is a concern that is not specific to reolink. So I don't think that replacing a device should be a reason not to pick this tbh

Reply:
I kind of dislike custom actions/services altogether, hence why I implemented the select entity in the previous PR.
because:
- they are harder to find in my opinion (don't show in the device page)
- targeting is not nice (as explained above)
- they require a lot more code to implement than just another entity_description in the list of entities.

Reply:
It has been reworked and now works both with device and with entities as targets (and a mix of multiple together).

It is become a hell of a lot of code for something which could have been just a simple select entity.....